### PR TITLE
docs: how to build or import a custom scalar

### PR DESCRIPTION
**Pull request by jfairley**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/docs.nestjs.com/blob/master/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] Docs
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

Adding documentation on how to register a custom scalar. This is different from the existing documentation that merely describes how to _override_ a built-in scalar. Fwiw, some of this is covered already in the "schema-first" section, but being there, I did not realize how relevant it was until I spent a lot of time doing outside research and troubleshooting. This adds more clarity for those choosing the "code-first" approach.

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

File: content/graphql/scalars.md [hunk 1]
```diff
@@ -75,6 +79,82 @@ Now we can use the `Date` type in our classes.
...
+To define a custom scalar, create a new `GraphQLScalarType` instance. We'll create a custom `UUID` scalar.
+
+```typescript
+const regex = /^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$/i;
+
+function validate(uuid: unknown): string | never {
+  if (typeof uuid !== "string") throw new Error("invalid uuid");
+  if (!regex.test(uuid)) throw new Error("invalid uuid");
+  return uuid;
+}
```
Comment (jmcdo29) [inline content/graphql/scalars.md:127]:
```suggestion
function validate(uuid: unknown): string | never {
  if (typeof uuid !== "string" || !regex.test(uuid)) {
    throw new Error("invalid uuid");
  }
  return uuid;
}
```

**Review by jmcdo29 — CHANGES_REQUESTED** [review]
Great addition! I've added on suggested change, but otherwise this looks good to me

**Comment by jfairley** [comment]
Thanks @jmcdo29. Suggestion committed.

**Review by jmcdo29 — APPROVED** [review]

**Review by Tony133 — APPROVED** [review]

**Comment by jfairley** [comment]
@jmcdo29 @Tony133 Is there anything else I need to do to get this merged?

**Comment by jmcdo29** [comment]
> @jmcdo29 @Tony133 Is there anything else I need to do to get this merged?

Right now, nope. Just gotta wait for @kamilmysliwiec to merge it

**Comment by kamilmysliwiec** [comment]
LGTM

**Pull request merged**